### PR TITLE
GH-66: Added contribution guidelines

### LINKED ISSUE #66: Add the contribution guidelines
As the project is a work in progress, there's a chance that the new contributors might show up who would be interested in working on this project. To communicate how people should contribute to this project, we should have some contribution guidelines. This will also help them verify that they're submitting well-formed pull requests and opening useful issues.

### PR DESCRIPTION
**Pull request by dmahajan980**
Resolves #66 

The format of our contribution guidelines is taken from the Infusion's [Contribution Guidelines](https://github.com/fluid-project/infusion/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+The Gamepad Navigator uses a workflow where individual contributors fork the project repository, work in a branch
+created off of `master`, and submit a pull request against the project repository's `master` branch to merge their
+contributions. See the
+[Coding and Commit Standards](https://wiki.fluidproject.org/display/fluid/Coding+and+Commit+Standards) for more clarity
+on the quality and standards of submitted contributions.
+
+### Open an Issue on GitHub
+
+The Gamepad Navigator uses [GitHub issues](https://github.com/fluid-lab/gamepad-navigator/issues) to keep track of the
```
Comment (the-t-in-rtf) [inline CONTRIBUTING.md:42]: You don't need "the" before open tickets, but I would simplify further, as in "GitHub to track issues".

You also need to add something equivalent to "All work should be associated with a specific GitHub issue."

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+(https://linkToAPI)
+```
+
+### Submit a Pull Request
+
+Once you've added tests with your code and passed the linting checks, you can submit a Pull Request (PR) to the
+repository's `master` branch. After the PR has been submitted, ping one or more maintainers on the "#fluid-work" IRC
+channel, fluid-work mailing list, or the PR itself. They will review the contribution, which typically results in a
+back and forth conversation and modifications to the PR. Merging into the project repo is a manual process and requires
+at least one maintainer to do that.
```
Comment (the-t-in-rtf) [inline CONTRIBUTING.md:114]: I see how you adapted the last sentence from the Infusion wording, but I think you shouldn't simplify what's being do as "to do that".  I would use the original wording to make it clear.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+All commits logs should include the following information:
+
+1. A reference to the GitHub issue this commit applies to (at the beginning of the first line).
+2. A short and meaningful summary of the commit, on the first line.
+3. A meaningful commit log describing the contents of the change. Also, mention if the pull request is meant to close a
+   ticket at the beginning. For example, "Resolves #18", "Fixes #28", et cetera.
+
+Sometimes, a commit may be trivial or entirely cosmetic (code reformatting, fixing typos in comments, et cetera).
+In those cases, it is acceptable to use the "NOGH:" prefix for your log. However, it is still required to provide a
```
Comment (the-t-in-rtf) [inline CONTRIBUTING.md:97]: OK, I see what you're going for, you can omit the requirement for a ticket in the previous section.  I would improve the last sentence, something like, "However, you must still provide...".

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+1. Search through the existing tickets to see if a similar ticket is already filed. If you find a ticket that seems to
+   match yours, add a comment to the same ticket describing any additional information you have (different context or
+   environment, speculated cause, possible solution, or any other feedback). Otherwise, create a new ticket.
+2. Provide as much relevant detail in the ticket description. If the ticket is a bug report, it should include the
+   following:
+   - Details of the actual behavior.
+   - Instructions for reproducing that bug.
+   - Difference between the actual and expected behavior.
+   - Additional details about the general context.
+3. Besides, more technical information can be added that might help whosoever working on that ticket (see below).
```
Comment (the-t-in-rtf) [inline CONTRIBUTING.md:64]: This needs a little work.  I would use "In addition" instead of "Besides", and avoid "technical" as they may be reporting design issues or mismatches with particular users (such as color or contrast issues).  I'd suggest "In addition, please provide any additional information that might help someone working on the ticket (see below)."

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+### Add Unit Tests
+
+Production-level code needs to be accompanied by a reasonable suite of unit tests. This helps others confirm that the
+code is working as intended, and allows the community to adopt more agile refactoring techniques while being more
+confident in our ability to avoid regressions. All unit tests should be written using jqUnit (check the
+[jqUnit documentation](https://docs.fluidproject.org/infusion/development/jqUnit.html) for more information).
+
+### Lint Your Code
+
+JavaScript is a highly dynamic and loose language, and many common errors are not picked up until run time. To avoid
```
Comment (the-t-in-rtf) [inline CONTRIBUTING.md:78]: "runtime" is more common.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+GH-29: Added bar method to inputMapper
+
+Resolves #29
+Refactored the inputMapper to include a new bar algorithm based on the latest changes in the Chrome API.
+(https://linkToAPI)
+```
+
+### Submit a Pull Request
+
+Once you've added tests with your code and passed the linting checks, you can submit a Pull Request (PR) to the
```
Comment (the-t-in-rtf) [inline CONTRIBUTING.md:110]: I would change "passed the linting checks" to "confirmed that the lint checks are passing".

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+<!-- TODO: Add an issue and pull request template on GitHub -->
+Go through the following steps to open a new issue/ticket:
+
+1. Search through the existing tickets to see if a similar ticket is already filed. If you find a ticket that seems to
+   match yours, add a comment to the same ticket describing any additional information you have (different context or
+   environment, speculated cause, possible solution, or any other feedback). Otherwise, create a new ticket.
+2. Provide as much relevant detail in the ticket description. If the ticket is a bug report, it should include the
+   following:
+   - Details of the actual behavior.
+   - Instructions for reproducing that bug.
```
Comment (the-t-in-rtf) [inline CONTRIBUTING.md:61]: I would say "the problem" as it may be a design flaw or failure to consider a range of use cases rather than a coding error.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+<!-- TODO: Add an issue and pull request template on GitHub -->
+Go through the following steps to open a new issue/ticket:
+
+1. Search through the existing tickets to see if a similar ticket is already filed. If you find a ticket that seems to
+   match yours, add a comment to the same ticket describing any additional information you have (different context or
+   environment, speculated cause, possible solution, or any other feedback). Otherwise, create a new ticket.
+2. Provide as much relevant detail in the ticket description. If the ticket is a bug report, it should include the
+   following:
+   - Details of the actual behavior.
```
Comment (the-t-in-rtf) [inline CONTRIBUTING.md:60]: I would simplify the wording here, it's a bit unnatural sounding.  I'd suggest combining with the third point, something like "What is happening?  What would you expect to happen instead?"

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+   - Instructions for reproducing that bug.
+   - Difference between the actual and expected behavior.
+   - Additional details about the general context.
```
Comment (the-t-in-rtf) [inline CONTRIBUTING.md:63]: "the general context" is overly broad.  I would suggest "about your environment" with examples in parentheses, such as "operating system, Chrome version, gamepad device details".

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+using the provided Grunt lint task (see below). Ensure that you run lint checks on your code before making a commit.
+
+``` snippet
+# Runs linting tasks
+grunt lint
+```
+
+### Create Meaningful Commit Logs
+
+All commits logs should include the following information:
```
Comment (the-t-in-rtf) [inline CONTRIBUTING.md:89]: "commits" shouldn't be plural, but I'd suggest "commit log messages" as that's what you really mean.

**Comment by dmahajan980** [comment]
@the-t-in-rtf thanks for the feedback. I've updated the pull accordingly. Kindly retake a look and suggest any further changes (if any).

File: CONTRIBUTING.md (hunk 9 above)
Comment (dmahajan980) [reply CONTRIBUTING.md]: Hi @the-t-in-rtf, I think "Browser details - name, version" would be more suitable instead of "Chrome version" , since the extension can run on multiple browsers.

Comment (the-t-in-rtf) [reply CONTRIBUTING.md]: It would be nice if we could avoid just saying "browser", as we don't want Safari or Firefox users to mistakenly assume that they can use it.  "Supported browser version (Chrome, Edge, Opera, and other Chromium-based)" is a bit wordy but perfectly clear.
